src/plugins/initiatives/initiativesDirectory.js

```diff
@@ -145,3 +147,56 @@ export async function _readFiles(
 
   return map;
 }
+
+// Checks the provided URL will parse and has no trailing slashes, search or hash.
+// Returns the validated URL without trailing slashes or throws.
+export function _validateUrl(remoteUrl: string): string {
+  try {
+    const url = new global.URL(remoteUrl);
+    if (url.search) {
+      throw `URL should not have a search component: ${url.search}`;
+    }
+    if (url.hash) {
+      throw `URL should not have a hash component: ${url.hash}`;
+    }
+    return url.toString().replace(/\/+$/, "");
+  } catch (e) {
+    throw new Error(
+      `Provided initiatives directory URL was invalid: ${remoteUrl}\n${e}`
+    );
+  }
+}
+
+// Takes a fileName, InitiativeFile map and converts it to Initiatives.
```

Comment:
nit: comment and argument names do not agree (fileName vs remoteUrl).
how about having it take a `InitiativesDirectory`? More descriptive at the type level, and then the reader can very clearly see what property we grab off it.

Reply:
Comment here wasn't intended to add a correct list of arguments. Updated it.
Also taking the `InitiativesDirectory` as argument.